File: haskell-server/src/EagerWords/Common/Message/StartGameRequest.hs

--
-- Copyright 2017-2020 Azad Bolour
-- Licensed under GNU Affero General Public License v3.0 -
--   https://github.com/azadbolour/eagerwords/blob/master/LICENSE.md
--

{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE DeriveAnyClass #-}

module EagerWords.Common.Message.StartGameRequest (
    StartGameRequest(..)
  )
  where

import GHC.Generics (Generic)
import Data.Aeson (FromJSON, ToJSON)
import Control.DeepSeq (NFData)

import EagerWords.Common.Domain.GameParams (GameParams)
import EagerWords.Common.Domain.Piece (Piece)
import EagerWords.Common.Domain.InitPieces (InitPieces)

data StartGameRequest = StartGameRequest {
    gameParams :: GameParams
  , initPieces :: InitPieces
  , userId :: String
} deriving (Eq, Show, Generic, NFData)

instance FromJSON StartGameRequest
instance ToJSON StartGameRequest

File: scala-server/src/main/scala/com/bolour/eagerwords/common/message/StartGameRequest.scala
/*
 * Copyright 2017-2020 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *    https://github.com/azadbolour/eagerwords.com/blob/master/LICENSE.md
 */
package com.bolour.eagerwords.common.message

import com.bolour.auth.common.domain.AuthEvidence
import com.bolour.eagerwords.common.domain.{GameParams, InitPieces}

/**
  * API request to start a game.
  *
  * @param loginEvidence Client id and token evidence of of the logged-in user. None for guest.
  * @param gameParams Basic specification of the game.
  * @param initPieces Initial pieces on the board - for testing.
  */
case class StartGameRequest(
  loginEvidence: Option[AuthEvidence],
  gameParams: GameParams,
  initPieces: InitPieces,
)
